WordPress/Classes/Services/SignupService.swift

```diff
@@ -0,0 +1,347 @@
+import Foundation
+import WordPressComAnalytics
+import WordPressShared
+
+
+/// Individual cases represent each step in the signup process.
+///
+enum SignupStatus: Int {
+    case Validating
+    case CreatingUser
+    case Authenticating
+    case CreatingBlog
+    case Syncing
+}
+
+
+typealias SignupStatusBlock = (status: SignupStatus) -> Void
+typealias SignupSuccessBlock = () -> Void
+typealias SignupFailureBlock = (error: NSError?) -> Void
+
+
+/// SignupService is responsible for creating a new WPCom user and blog.
+/// The entry point is `createBlogAndSigninToWPCom` and the service takes care of the rest.
+///
+public class SignupService : LocalCoreDataService
+{
+    private let LanguageIDKey = "lang_id"
+    private let BlogDetailsKey = "blog_details"
+    private let BlogNameLowerCaseNKey = "blogname"
+    private let BlogNameUpperCaseNKey = "blogName"
+    private let XMLRPCKey = "xmlrpc"
+    private let BlogIDKey = "blogid"
+    private let URLKey = "url"
+
+
+    /// Starts the process of creating a new blog and wpcom account for the user. 
+    /// 
+    /// - Parameters:
+    ///     - url: The url for the new wpcom site
+    ///     - blogTitle: The title of the blog
+    ///     - emailAddress: The user's email address
+    ///     - username: The desired username
+    ///     - password: The user's password
+    ///     - status: The status callback
+    ///     - success: A success calback
+    ///     - failure: A failure callback
+    ///
+    func createBlogAndSigninToWPCom(blogURL url:String,
+                                            blogTitle: String,
+                                            emailAddress: String,
+                                            username: String,
+                                            password: String,
+                                            status: SignupStatusBlock,
+                                            success: SignupSuccessBlock,
+                                            failure: SignupFailureBlock) {
+
+        let params = SigninParams(url: url, title: blogTitle, email: emailAddress, username: username, password: password)
+        validateWPComBlogWithParams(params, status: status, success: success, failure: failure)
```

Comment:
My main concern about the new service is that you are calling a `validate` method that "Validates that the blog can be created and is not already taken", but it will actually create the account and the site.

I think the request chaining should be done in this method and not in the individual requests. It'll look a bit terrible, but it's better than having methods doing more things than what they promise.